> > > Hopefully
> > 
> > 
> > Does that mean you didn't check if it fixes it? I think we should understand how the test passed without this PR before including that.
> 
> I do not have the right machines, etc. to test it. I have asked some friends at Apple to test it for me, but have not heard back yet (but I only just posted the PR).

I applied this patch to a current TOT checkout I had, and ran the test suite.  The only failure I got was:

```
FAIL: LLDB (/Users/jingham/Work/LLDB/llvm-dot-org/scratch-lldb/all-in-one/bin/clang-arm64) :: test_frame_recognizer (TestLibcxxInternalsRecognizer.LibCxxInternalsRecognizerTestCase)
======================================================================
FAIL: test_frame_recognizer (TestLibcxxInternalsRecognizer.LibCxxInternalsRecognizerTestCase)
   Test that implementation details of libc++ are hidden
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jingham/Work/LLDB/llvm-dot-org/scratch-lldb/llvm-project/lldb/packages/Python/lldbsuite/test/decorators.py", line 151, in wrapper
    return func(*args, **kwargs)
  File "/Users/jingham/Work/LLDB/llvm-dot-org/scratch-lldb/llvm-project/lldb/test/API/lang/cpp/libcxx-internals-recognizer/TestLibcxxInternalsRecognizer.py", line 69, in test_frame_recognizer
    self.assertIn(expected_parent, func_name)
AssertionError: '::sort' not found in 'bool std::__1::is_sorted[abi:de220000]<int*, std::__1::__debug_less<bool (*)(int, int)>>(int*, int*, std::__1::__debug_less<bool (*)(int, int)>)'
Config=arm64-/Users/jingham/Work/LLDB/llvm-dot-org/scratch-lldb/all-in-one/bin/clang
----------------------------------------------------------------------
Ran 1 test in 2.809s

```

But I'm pretty sure that's not you.

--- a/lldb/source/ValueObject/DILEval.cpp
+++ b/lldb/source/ValueObject/DILEval.cpp
@@ -348,45 +348,46 @@ Interpreter::Visit(const ArraySubscriptNode *node) {
 
     if (!m_use_synthetic && is_objc_pointer) {
       std::string err_msg =
-          llvm::formatv("\"(%s) %s\" is an Objective-C pointer, and cannot be "
+          llvm::formatv("\"({0}) {1}\" is an Objective-C pointer, and cannot be "
                         "subscripted",
                         base->GetTypeName().AsCString("<invalid type>"),
                         var_expr_path_strm.GetData());
       return llvm::make_error<DILDiagnosticError>(m_expr, std::move(err_msg),
                                                   node->GetLocation());
-    } else if (is_objc_pointer) {
+    }
+    if (is_objc_pointer) {
       lldb::ValueObjectSP synthetic = base->GetSyntheticValue();
       if (!synthetic || synthetic == base) {
         std::string err_msg =
-            llvm::formatv("\"(%s) %s\" is not an array type",
+            llvm::formatv("\"({0}) {1}\" is not an array type",
                           base->GetTypeName().AsCString("<invalid type>"),
                           var_expr_path_strm.GetData());
         return llvm::make_error<DILDiagnosticError>(m_expr, std::move(err_msg),
                                                     node->GetLocation());
-      } else if (static_cast<uint32_t>(child_idx) >=
-                 synthetic->GetNumChildrenIgnoringErrors()) {
+      }
+      if (static_cast<uint32_t>(child_idx) >=
+          synthetic->GetNumChildrenIgnoringErrors()) {
         std::string err_msg = llvm::formatv(
-            "array index %ld is not valid for \"(%s) %s\"", child_idx,
+            "array index {0} is not valid for \"({1}) {2}\"", child_idx,
             base->GetTypeName().AsCString("<invalid type>"),
             var_expr_path_strm.GetData());
         return llvm::make_error<DILDiagnosticError>(m_expr, std::move(err_msg),
                                                     node->GetLocation());
-      } else {
-        child_valobj_sp = synthetic->GetChildAtIndex(child_idx);
-        if (!child_valobj_sp) {
-          std::string err_msg = llvm::formatv(
-              "array index %ld is not valid for \"(%s) %s\"", child_idx,
-              base->GetTypeName().AsCString("<invalid type>"),
-              var_expr_path_strm.GetData());
-          return llvm::make_error<DILDiagnosticError>(
-              m_expr, std::move(err_msg), node->GetLocation());
-        }
-        if (m_use_dynamic != lldb::eNoDynamicValues) {
-          if (auto dynamic_sp = child_valobj_sp->GetDynamicValue(m_use_dynamic))
-            child_valobj_sp = std::move(dynamic_sp);
-        }
-        return child_valobj_sp;
       }
+      child_valobj_sp = synthetic->GetChildAtIndex(child_idx);
+      if (!child_valobj_sp) {
+        std::string err_msg = llvm::formatv(
+            "array index {0} is not valid for \"({1}) {2}\"", child_idx,
+            base->GetTypeName().AsCString("<invalid type>"),
+            var_expr_path_strm.GetData());
+        return llvm::make_error<DILDiagnosticError>(
+            m_expr, std::move(err_msg), node->GetLocation());
+      }
+      if (m_use_dynamic != lldb::eNoDynamicValues) {
+        if (auto dynamic_sp = child_valobj_sp->GetDynamicValue(m_use_dynamic))
+          child_valobj_sp = std::move(dynamic_sp);
+      }
+      return child_valobj_sp;
     }
 
     child_valobj_sp = base->GetSyntheticArrayMember(child_idx, true);

--- a/lldb/source/ValueObject/DILEval.cpp
+++ b/lldb/source/ValueObject/DILEval.cpp
@@ -347,12 +347,10 @@ Interpreter::Visit(const ArraySubscriptNode *node) {
       is_objc_pointer = false;
 
     if (!m_use_synthetic && is_objc_pointer) {
-      std::string err_msg =
-          llvm::formatv(
-              "\"({0}) {1}\" is an Objective-C pointer, and cannot be "
-              "subscripted",
-              base->GetTypeName().AsCString("<invalid type>"),
-              var_expr_path_strm.GetData());
+      std::string err_msg = llvm::formatv(
+          "\"({0}) {1}\" is an Objective-C pointer, and cannot be subscripted",
+          base->GetTypeName().AsCString("<invalid type>"),
+          var_expr_path_strm.GetData());
       return llvm::make_error<DILDiagnosticError>(m_expr, std::move(err_msg),
                                                   node->GetLocation());
     }

--- a/lldb/source/ValueObject/DILEval.cpp
+++ b/lldb/source/ValueObject/DILEval.cpp
@@ -348,10 +348,11 @@ Interpreter::Visit(const ArraySubscriptNode *node) {
 
     if (!m_use_synthetic && is_objc_pointer) {
       std::string err_msg =
-          llvm::formatv("\"({0}) {1}\" is an Objective-C pointer, and cannot be "
-                        "subscripted",
-                        base->GetTypeName().AsCString("<invalid type>"),
-                        var_expr_path_strm.GetData());
+          llvm::formatv(
+              "\"({0}) {1}\" is an Objective-C pointer, and cannot be "
+              "subscripted",
+              base->GetTypeName().AsCString("<invalid type>"),
+              var_expr_path_strm.GetData());
       return llvm::make_error<DILDiagnosticError>(m_expr, std::move(err_msg),
                                                   node->GetLocation());
     }
@@ -380,8 +381,8 @@ Interpreter::Visit(const ArraySubscriptNode *node) {
             "array index {0} is not valid for \"({1}) {2}\"", child_idx,
             base->GetTypeName().AsCString("<invalid type>"),
             var_expr_path_strm.GetData());
-        return llvm::make_error<DILDiagnosticError>(
-            m_expr, std::move(err_msg), node->GetLocation());
+        return llvm::make_error<DILDiagnosticError>(m_expr, std::move(err_msg),
+                                                    node->GetLocation());
       }
       if (m_use_dynamic != lldb::eNoDynamicValues) {
         if (auto dynamic_sp = child_valobj_sp->GetDynamicValue(m_use_dynamic))

